library/alloc/src/collections/btree/map.rs

```diff
@@ -178,6 +178,8 @@ pub struct BTreeMap<
     length: usize,
     /// `ManuallyDrop` to control drop order (needs to be dropped after all the nodes).
     pub(super) alloc: ManuallyDrop<A>,
+    // For dropck; the `Box` avoids making the `Unpin` impl more strict than before
+    _marker: PhantomData<crate::boxed::Box<(K, V)>>,
```

Comment:
I think this is yet another reminder for us that we need something more intuitive than PhantomData. PhantomData is used to control Send, Sync, dropck, UnwindSafe, UnwindSafeRef, variance, and Unpin. Possibly more (in the future). It happens so often that we try to control just some of these, and accidentally end up affecting one of the others too. (Like what would've happened here if (K, V) wasn't wrapped in a Box.)

cc @rust-lang/lang

Reply:
This reminds me of a crate that I wrote half a year ago. https://docs.rs/phtm/1.0.3/phtm/